React/Profiler/RCTProfile.m

```diff
@@ -42,8 +42,7 @@
 
 #pragma mark - Variables
 
-// This is actually a BOOL - but has to be compatible with OSAtomic
-static volatile uint32_t RCTProfileProfiling;
+static atomic_bool RCTProfileProfiling = ATOMIC_VAR_INIT(NO);
```

Comment:
Hmm `constexpr` isn't available except in C++. We can make this `RCTProfile.mm` but that's a bigger change and we'll lose a tiny bit of efficiency (due to ARC exception safety). What do?